bin/rose-make-docs

```diff
@@ -0,0 +1,176 @@
+#     `strict`
+#         Runs sphinx-build in a "strict" mode where cashing is turned off and
+#         warnings are turned into errors.
+#     `linkcheck`
+#         Checks external links.
+#
+#-----------------------------------------------------------------------------
+set -e
+set -o pipefail
+
+# Move into rose directory
+cd "$(dirname $0)/../"
+
+VENV_PATH='venv'  # Path for virtualenv.
+USING_VENV=false  # Set to `true` when the virtualenv is being used.
+SPHINX_PATH=sphinx  # Path to the sphinx directory.
+
+venv-activate () {
+    USING_VENV=true
+    . "${VENV_PATH}/bin/activate"
+}
+
+venv-install () {
+    venv-destroy
+    virtualenv --python=python2.7 "${VENV_PATH}"
+    venv-activate
+    pip install sphinx
+    pip install sphinx_rtd_theme
+}
+
+venv-deactivate () {
+    deactivate >/dev/null 2>&1 || true
+}
+
+venv-destroy () {
+    venv-deactivate
+    rm -rf "${VENV_PATH}"
+}
+
+# Parse command line args.
+VENV_MODE=false
+DEV_MODE=${SPHINX_DEV_MODE:=false}
+BUILDS=''
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        --venv)
+            VENV_MODE=true
+            shift
+            ;;
+        --dev)
+            DEV_MODE=true
+            shift
+            ;;
+        *)
+            BUILDS="$BUILDS $1"
+            shift
+            ;;
+    esac
+done
+if [[ -z "${BUILDS}" ]]; then
+    BUILDS='html'
+fi
+
+# Use virtualenv if present and requested or if we are in development mode.
+if [[ "${DEV_MODE:-false}" || "${VENV_MODE:-false}" ]]; then
+    if [ -d "${VENV_PATH}" ]; then
+        venv-activate
+    fi
+fi
+
+# Check optional builders.
+for build in ${BUILDS}; do
+    if ! rose-check-software-docs ${build} >/dev/null; then
+        echo "Software dependencies for a ${build} build are missing."
+        echo "run `rose software-check-docs ${build}` for details."
+        exit 1
+    fi
+done
+
+# Check core (sphinx) builder.
+if ! rose-check-software-docs core >/dev/null; then
+    if ${VENV_MODE}; then
+        venv-install
+    else
+        echo 'Software required by the rose documentation builder is not'
+        echo 'present (run `rose software-check-docs` for details).'
+        echo
+        echo 'The documentation can still be built by installing dependencies'
+        echo 'in a python virtual enironment (virtualenv). This environment'
+        echo "will be created in rose/${VENV_PATH} and will be removed after"
+        echo 'use. To keep the virtualenv for future builds run this command'
+        echo 'with the "--dev" option.'
+        echo
+        while read -p 'Proceed using a virtualenv (y/n)? ' usr; do
+            case $usr in
+                [Yy])
+                    venv-install
+                    break
+                    ;;
+                [Nn])
+                    echo
+                    echo 'Unable to build documentation.'
+                    exit 1
+                    ;;
+            esac
+        done
+    fi
+fi
+
+# Run sphinx make.
```

Comment:
This is purposeful, the `BUILDS` variable contains space separated variables which should be interpreted as arguments rather than passing the whole variable `BUILD` as a single argument.